CI/travis/check_kernel.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -e
+
+KERNEL_TYPES="/tmp/mainline_types.h"
+IIOH="./iio.h"
+CHANNELC="./channel.c"
+
+if [ ! -f ${IIOH} ] ; then
+	echo can not find ${IIOH}
+	exit 1
+fi
+
+if [ ! -f ${CHANNELC} ] ; then
+	echo can not find ${CHANNELC}
+	exit 1
+fi
+
+rm -f ${KERNEL_TYPES}
+wget -O ${KERNEL_TYPES} https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/plain/include/uapi/linux/iio/types.h
+
+ret=0
+
+for enum in iio_chan_type iio_modifier
```

Comment:
what if we change it a bit, to take the contents of `include/uapi/linux/iio/types.h` between `#define _UAPI_IIO_TYPES_H_` and  `#endif /* _UAPI_IIO_TYPES_H_ */` and dump it in `iio.h` between 2  comment markers

something like:
```
......
struct iio_context_info;
struct iio_scan_context;

/* --- added automatically from 'include/uapi/linux/iio/types.h' */

contents of include/uapi/linux/iio/types.h

/* --- end 'include/uapi/linux/iio/types.h' */
```

we could then use  git diff to check if we need to update something;
we could also use the same logic to just update the header [for lazy people like me];

i know there are a few minor downsides [in the beginning]:
* some things do need to be reworked a bit though; like for example, we would need to change `IIO_CHAN_TYPE_UNKNOWN` to be a `#define IIO_CHAN_TYPE_UNKNOWN  UINT_MAX` or something; this would sit outside of that header-dump for a while, but we could also propose that field upstream into the enum;
* we would also get some event definitions; but i also don't see this as a big problem, as it just simplifies a few things when we add event support to libiio

atm, there's only 2 headers for IIO in `include/uapi/linux/iio/`: `types.h` & `events.h` ; assuming things will grow [in there], this sync-mechanism will be a bit more handy & scalable [1]

bonus-mention [related to point [1] ] : the AXI ADC driver has been partially upstreamed [rrecently], and we're currently working to get more high-speed buffer support mainlined; and then the AXI DAC;
with high-speed buffer support [ioctl definitions needing to go in there] this sync mechanism could also be handy

Reply:
Right now - I didn't want to change any of the libiio source - since things are changed so infrequently - I thought checking and error would be fine. When we support events - we can look at re-working things then as part of that.

Reply:
ok;
fine by me